packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -132,7 +132,7 @@ String buildAotSnapshot(
     '--no-sim-use-hardfp',
   ];
 
-  if (!(tools.engineRelease || buildMode == BuildMode.release)) {
+  if (buildMode != BuildMode.release) {
```

Comment:
I'm confused here. We're passing in `--no-checked` for profile and debug builds?